contracts/security-token/RestrictedTransferAgent.sol

```diff
@@ -24,7 +23,16 @@ contract RestrictedTransferAgent is SecurityTransferAgent {
       return value;
     }
 
-    if (KYC.isWhitelisted(to)) {
+    // The following ifs might seem counterintuitive, but this function will
+    // be executed upon every transfer, so we want to make sure that despite of
+    // Solidity's optimization settings, only relevant external calls are made
+    if (KYC.getFlag(to, 0)) {
+      if (KYC.getFlag(from, 0)) {
```

Comment:
this could be combined with the code above as `KYC.getFlag(to, 0) && KYC.getFlag(from, 0)`

Reply:
Yes, the problem is that I haven't had time to study what kind of bytecode that would result: would both of the functions be called anyway. Hence the comment above

Reply:
In most programming languages, whenever the first expression in an `and`(&&) operation evaluates to `false`, the second is never evaluated. I just checked this by writing some dummy code in remix:
```
pragma solidity ^0.4.24;

contract test3 {
    uint232 public count;
    
    constructor() public {
        count = 0;
    }
    
    function check () public returns (bool){
        if (increment() && increment()) {
            return true;
        }
        return false;
    }
    function increment() internal returns (bool){
        count = count + 1;
        return false;
    }
    
}

```
from the test I can infer that solidity is no different.

Reply:
Yes, but I would, and will not presume anything when it comes to smart contracts, thanks for proving it, though :slightly_smiling_face: